src/modules/site/base/views/user.py

```diff
@@ -68,3 +105,22 @@ def user_update():
     user.save()
     return redirect(url_for('user.user_profile'))
   return render_template('user/update.html', **locals())
+
+
+@user_bp.route("/password/reset", methods=["GET", "POST"])
+@jwt_required()
+def user_reset_password():
+  title = "Reset Password"
+  disable_parent_breadcrumb = True
+  user = get_current_user()
+  form = PasswordResetForm(request.form)
+
+  if request.method == 'POST' and form.validate():

```

Comment:
The more I see this pattern, the more I want it to read.

```
if not (request.method == "POST" and form.validate()):
    return # render template/etc.
# (...)
return redirect(...)
```
Thinking of ways to codify this syntax preference for if statements -- taking a stab at it: _always prefer the version of the code with the least amount of indentation levels (within reason) that achieve the same behavior._

Reply:
In other words, always prefer to have less code indented if possible?

The problem I see though is that we don't have a fall through if we short circuit if some cases like this one:
```
@user_bp.route("/recover", methods=["GET", "POST"])
def user_recover():
  """ Enter an email address for a user to send a password reset link """
  title = "Recover User"
  disable_parent_breadcrumb = True
  form = RecoverUserForm(request.form)

  if request.method == 'POST' and form.validate():
    email = request.form.get("email")
    users = get_local_user_by_email(email)

    if not users or not users[0]:
      # Flash info message even if email fails in order to protect from attempts at finding valid emails
      flash(f"Please check your email ({email}) for a password reset link.", 'info')
      logger.error(f"Failed to retrieve LOCAL user for email: {email}")
      return redirect('/')
    
    user = users[0]
    token = create_one_time_token(id=user.get('username'), destination=url_for('user.user_reset_password'))
    password_reset_link = url_for('auth.token', token=token, _external=True)
    try:
      send_email({
        "from": "no-reply@elegansvariation.org",
        "to": [ email ],
        "subject": "CeNDR Password Reset",
        "text": PASSWORD_RESET_EMAIL_TEMPLATE.format(email=email, password_reset_link=password_reset_link)
      })
      logger.info(f"Sent password reset email: {email}, link: {password_reset_link}")
    except Exception as err:
      flash('Unable to reset your credentials at this time. Please try again later.', 'error')
      logger.error(f"Failed to send email: {err}")
    
  return render_template('user/recover_user.html', **locals())
```